src/commands/openapi/refs.ts

```diff
@@ -0,0 +1,323 @@
+/* eslint-disable no-param-reassign */
+import type { OASDocument } from 'oas/types';
+import type { IJsonSchema } from 'openapi-types';
+
+import fs from 'node:fs';
+import path from 'node:path';
+
+import { Args, Flags } from '@oclif/core';
+import chalk from 'chalk';
+import prompts from 'prompts';
+
+import analyzeOas from '../../lib/analyzeOas.js';
+import BaseCommand from '../../lib/baseCommand.js';
+import { titleFlag, workingDirectoryFlag } from '../../lib/flags.js';
+import { warn, debug } from '../../lib/logger.js';
+import prepareOas from '../../lib/prepareOas.js';
+import promptTerminal from '../../lib/promptWrapper.js';
+import { validateFilePath } from '../../lib/validatePromptInput.js';
+
+type SchemaCollection = Record<string, IJsonSchema>;
+
+export default class OpenAPISolvingCircularityAndRecursiveness extends BaseCommand<
+  typeof OpenAPISolvingCircularityAndRecursiveness
+> {
+  static description =
+    'Resolves circular and recursive references in OpenAPI by replacing them with object schemas. Not all circular references can be resolved automatically.';
```

Comment:
can you add a `summary` field to this class that contains just this part of this description:

> Resolves circular and recursive references in OpenAPI by replacing them with object schemas.